client/src/components/Search/Filters/FilterTypes/DateFilter/DateFilter.js

```diff
@@ -0,0 +1,172 @@
+import * as React from 'react';
+import DatePicker from 'react-datepicker';
+import { Form } from 'react-bootstrap';
+import * as moment from 'moment';
+import './DateFilter.scss';
+
+export default class DateFilter extends React.Component {
+    constructor(props) {
+        super(props)
+        const { data } = props;
+        this.state = {
+            params:{
+                decade: data.params.decade || null,
+                start: data.params.start || null, //TODO - make date dynamic
+                end: data.params.end || null
+            },
+
+            filterMode: data.config.filterMode || 'decade',
+            dateFilterStart: new Date(),
+            dateFilterEnd: new Date(),
+            filterList: []
+        };
+    }
+
+    async handleFilterChange(name, value) {
+
+        const updatedState = {};
+        const {onChange} = this.props;
+        updatedState[name] = value;
+        await this.setState({
+            params:{
+                ...this.state.params,
+                ...updatedState
+            }
+        });
+        onChange(this.state.params)
+    }
+
+    async handleDateFilter(type, value) {
+        switch (type) {
+            case 'decade':
```

Comment:
define these strings with an enum constant and then use the enum everywhere these are used